truffle/src/main/java/org/truffleruby/parser/BodyTranslator.java

```diff
@@ -1729,7 +1729,7 @@ public RubyNode visitGlobalAsgnNode(GlobalAsgnParseNode node) {
             rhs = new CheckOutputSeparatorVariableTypeNode(rhs);
             rhs.unsafeSetSourceSection(sourceSection);
         } else if (name.equals("$_")) {
-            if (getSourcePath(sourceSection).endsWith(buildPartialPath("truffle", "rubysl", "rubysl-stringio", "lib", "rubysl", "stringio", "stringio.rb"))) {
+            if (getSourcePath(sourceSection).endsWith(buildPartialPath("rubysl", "rubysl-stringio", "lib", "rubysl", "stringio", "stringio.rb"))) {
```

Comment:
(Unrelated) This seems a bit too much, let's hard-code the `/` as anyway that is what is visible at the Ruby level.